From a1d7c582392c8bc87fa9411af77b20e011944357 Mon Sep 17 00:00:00 2001
From: Alexander Kanavin <alex.kanavin@gmail.com>
Date: Thu, 25 Jan 2018 17:55:02 +0200
Subject: [PATCH] gst/gstpluginloader.c: when env var is set do not fall
 through to system plugin scanner

If we set a custom GST_PLUGIN_SCANNER env var, then we probably want to use that and only that.

Falling through to the one installed on the system is problamatic in cross-compilation
environemnts, regardless of whether one pointed to by the env var succeeded or failed.

Upstream-Status: Pending
Signed-off-by: Alexander Kanavin <alex.kanavin@gmail.com>
---
 gst/gstpluginloader.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/gst/gstpluginloader.c b/gst/gstpluginloader.c
index 430829d..3a75731 100644
--- a/gst/gstpluginloader.c
+++ b/gst/gstpluginloader.c
@@ -471,9 +471,7 @@ gst_plugin_loader_spawn (GstPluginLoader * loader)
     helper_bin = g_strdup (env);
     res = gst_plugin_loader_try_helper (loader, helper_bin);
     g_free (helper_bin);
-  }
-
-  if (!res) {
+  } else {
     GST_LOG ("Trying installed plugin scanner");
 
 #ifdef G_OS_WIN32
-- 
2.15.1

